### ISSUE TITLE
Using read_file with a string (non-file-like-object) causes confusing exceptions

### ISSUE THREAD

**Issue by cmorty** [pyscaffold/configupdater#5]
Calling `read_file` with a string will cause a `MissingSectionHeaderError` exception. (I know docs say otherwise, it's very confusing none the less.)

**Comment by FlorianWilhelm** [comment]
Thanks, can you provide some code to reproduce and to see what you want to do?

**Comment by cmorty** [comment]
Hey, note that I changed the bug. The first version was a PBKAC.

```
from configupdater import ConfigUpdater

updater = ConfigUpdater()
updater.read_file("test.cfg")
```
causes

```
configparser.MissingSectionHeaderError: File contains no section headers.
file: '<???>', line: 1
't'
```

The acutual error is that `test.cfg` is not a file-like object.

**Comment by FlorianWilhelm** [comment]
@cmorty Thanks for your patience, this is fixed in version 1.0.1.
